lib/ui/pages/write_page.dart

```diff
@@ -158,3 +167,37 @@ QuillController createQuillController(Idea idea) {
     selection: const TextSelection.collapsed(offset: 0),
   );
 }
+
+/// A class for periodically saving text from Quill to the database.
+///
+/// Call [Saver.notify] when there are changes to the idea's text.
+/// [Saver] will wait and accumulate changes for [Saver.delay] and then save.
+/// Calling [Saver.notify] when it is already waiting has no effect.
+///
+/// Call [Saver.save] to save immediately and cancel waiting for changes.
+class Saver {
+  final String userId;
+  final String ideaId;
+  final QuillController controller;
+  static const delay = Duration(seconds: 2);
+
+  Timer? _timer;
+
+  Saver(this.userId, this.ideaId, this.controller);
+
+  void notify() {
+    if (_timer == null || !_timer!.isActive) {
```

Comment:
Idea - to further reduce bandwidth usage & spam, we could make it debounced - only after there has been no activity after X seconds, we save.
```
    if (_timer != null) {
      _timer!.cancel();
    }
    _timer = Timer(delay, save);
```